src/powfacpy/case_studies.py

```diff
@@ -219,29 +227,89 @@ def set_parameters(self, case_obj_or_case_num):
       if value is not None:
         self.set_attr_by_path(path, value)  
 
-  def get_study_cases(self, conditions):
+  def get_study_cases(self, conditions, return_case_numbers=False):
     """Retrieve study case objects depending on parameter values.
     Arguments:
-      conditions: A dictionary with
-        keys: parameter names
-        values: lambda function with boolean return value depending on 
-          parameter (key)
+      conditions: 
+        Either a dictionary with
+          keys: parameter names
+          values: lambda function with boolean return value depending on 
+            parameter (key)
+        or a lambda function with an iterable containing all parameters.
+          Example: lambda x: x[0] >= 2 and x[2] == 'A'
+          Note that the order of the parameters in x must be the same as the
+          order of the keys in self.parameter_values.
     Returns the study case objects whose parameters fullfill the conditions.
+    If return_case_numbers is True, a tuple including the case numbers is returned.
 
-    Example:
+    Example 1:
       get_study_cases({"par1": lambda x: x == 2, "par2": lambda x: x>0})
         This returns the study cases for which 'par1' equals 2 and 'par2' is 
-        positive.       
+        positive. 
+    Example 2 (lambda function): 
+      get_study_cases(lambda x: x[0] >= 2 and x[2] == 'A')          
+    """
```

Comment:
Looks good! But I did not follow all the details now. Can be merged if the tests run.

Reply:
I think you need to approve ("At least 1 approving review is required by reviewers")